### PR TITLE
Use RemoteNode.connection_info for TCP connection

### LINKED ISSUE #2694: Private IP address is not used in test verify_stop_start_in_platform
## What
When `use_public_address` is set to `false` then test `verify_stop_start_in_platform` still uses public_address of Azure VMs (hence test fails in environment where only private_address are accessible)

## Why
When `use_public_address` is set to `false` then function `set_connection_info` in RemoteNode class only sets private_ip in `self._connection_info`  but variable self.public_address still points to public ip which is later used by test cases.

## Error log:
Log where is issue is reproduced: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=339536&view=logs&j=528a33a8-464e-565a-7855-df1244039b47&t=86425f9e-1e14-58f6-d288-a39ec4000eff&l=1845 

The above pipeline runs in an environment where only private_ip are accessible. The log says that connection timeout to ip `13.84.212.170:22` (which is a public ip)  while the private ip is `10.0.0.5:22`

### PR DESCRIPTION
**Pull request by xordux**
Fixes #2694

Tested it in [this pipeline](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=344126&view=results).

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -2167,15 +2167,20 @@ def _enable_ssh_on_windows(self, node: Node) -> None:
 
         remote_node = cast(RemoteNode, node)
 
+        node_ssh_port = remote_node.connection_info[
```
Comment (xordux) [inline lisa/sut_orchestrator/azure/platform_.py:2123]: using `remote_node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_PORT]` in log.debug below was creating very long line and hence failing flake8 test. That's I created this variable and used it in log.debug statement below.

File: lisa/tools/reboot.py [hunk 2]
```diff
@@ -166,16 +167,23 @@ def reboot(self, time_out: int = 600) -> None:
...
+                    ],
+                    port=remote_node.connection_info[
+                        constants.ENVIRONMENTS_NODES_REMOTE_PORT
+                    ],
                     log=self._log,
                     timeout=20,
                 )
 
                 if not connected:
+                    node_ssh_port = remote_node.connection_info[
```
Comment (xordux) [inline lisa/tools/reboot.py:181]: using `remote_node.connection_info[constants.ENVIRONMENTS_NODES_REMOTE_PORT]` in LisaException below was creating very long line and hence failing flake8 test. That's I created this variable and used it in LisaException statement below.

**Comment by xordux** [comment]
Tested it in [this pipeline](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=344126&view=results).

While it fixes the issue for tests like:
- Storage.hot_add_disk_parallel
- Storage.hot_add_disk_parallel_standard_ssd
- Storage.hot_add_disk_parallel_premium_ssd
- Sriov.verify_sriov_disable_enable 
- and some more ..

There are still some tests which are still using public ip, like:
- Provisioning.verify_stop_start_in_platform
- Sriov.verify_sriov_add_max_nics
- Synthetic.synthetic_provision_with_max_nics_stop_start_from_platform_validation
- and some more ..
More changes are needed to fix them all.

**Comment by squirrelsc** [comment]
@xordux you are welcome to continue to work on it, we don't need to enable for all cases in one PR.

**Comment by xordux** [comment]
I didn't got the notification for you comment @squirrelsc , I saw it yesterday only. Now I've re-opened this PR. Thank you.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM. We need to convert connection_info to be an object. So it's easier to reference.

**Review by LiliDeng — APPROVED** [review]

**Comment by LiliDeng** [comment]
> @LiliDeng LGTM. We need to convert connection_info to be an object. So it's easier to reference.

I have created a task to track this.

**Pull request merged**